backend/src/atlas_assistant/api.py

```diff
@@ -106,3 +167,74 @@ async def login(
             detail="Incorrect username or password",
             headers={"WWW-Authenticate": "Bearer"},
         )
+
+
+@app.post(
+    "/chat", tags=["chat"], response_model=ToolResponseMessage | AiResponseMessage
+)
+async def chat(
+    chat_request: ChatRequest,
+    settings: Annotated[Settings, Depends(get_settings)],
+    current_user: Annotated[User, Depends(get_current_user)],  # pyright: ignore[reportUnusedParameter]
+    accept: Annotated[str | None, Header()] = None,
+) -> StreamingResponse:
+    agent = create_agent(settings)
+    thread_id = uuid.uuid4()
+    event_stream = (accept and "text/event-stream" in accept) or False
+    return StreamingResponse(
+        query_agent(agent, chat_request.query, str(thread_id), settings, event_stream),
+        media_type="text-event-stream" if event_stream else "application/x-ndjson",
+    )
+
+
+@app.exception_handler(GraphRecursionError)
+async def graph_recursion_handler(
+    _request: Request, exc: GraphRecursionError
+) -> JSONResponse:
+    return JSONResponse(
+        status_code=status.HTTP_400_BAD_REQUEST,
+        content=Error(message=str(exc), type="graph recursion"),
+    )
```

Comment:
I saw this happen once, so I added a test for it. Like many things when working with these bots, robust testing (if we have time for that) will be a PITA.